ion-schema/src/isl/ranges.rs

```diff
@@ -1,28 +1,46 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Range types used in Ion Schema.
+//!
+//! ### Why so many range types?
+//! - [`UsizeRange`] is used for `*_length` constraints and `occurs`
+//! - [`U64Range`] is used for the `precision` constraint because [`Decimal`] precision is measured
+//!   using `u64`. It would be ideal to use [`NonZeroU64`][std::num::NonZeroU64] for this case, but
+//!   `NonZeroU64` is difficult to work with because it doesn't implement core ops (such as `Add`)
+//!   and as a result, it cannot even implement the checked ops traits (such as `CheckedAdd`) from
+//!   the `num_traits` crate. See https://github.com/rust-num/num-traits/issues/274.
```

Comment:
The "It would be ideal [...]" portion of this seems like it might be better suited to a developer comment, but I don't feel strongly about it.